core/src/main/scala/persistentscheduler/PersistentScheduler.scala

```diff
@@ -78,48 +80,47 @@ class PersistentScheduler(persistence: SchedulerPersistence) extends Actor
     self ! State(subscriptions)
   }
 
-  def schedulePersistenceCheckForEvents(): Unit = {
+  private def schedulePersistenceCheckForEvents(): Unit = {
     scheduler.schedule(1.minute, 1.minute, self, CheckPersistenceForEvents())
   }
 
-  def scheduleNextEventFromPersistence(): Unit = {
+  private def scheduleNextEventFromPersistence(): Unit = {
     nextCancellable.filter(!_.isCancelled).foreach(_.cancel())
 
     nextEvent = persistence.next(1).asScala.headOption
-    nextCancellable = nextEvent.map(schedule(_))
+    nextCancellable = nextEvent.flatMap(schedule(_).toOption)
```

Comment:
the `schedule` method could return an Option, then you don't need the conversions everywhere